src/serde.rs

```diff
@@ -47,3 +50,51 @@ impl<V: 'static> Fields<V> {
             .flatten()
     }
 }
+
+/// Enables deserializing from a value that could be a bool or uri or a struct.
+/// Example:
+/// healthcheck: bool
+/// healthcheck: "...."
+/// healthcheck.enabled: bool
+/// Both are accepted.
+pub fn bool_or_str_or_struct<'de, T, D>(deserializer: D) -> Result<T, D::Error>
```

Comment:
If you keep a `healthcheck_uri` option, the serde bits could be simplified to only need a `bool_or_struct` deserializer.